blueflood-elasticsearch/src/test/java/com/rackspacecloud/blueflood/io/ElasticIOIntegrationTest.java

```diff
@@ -157,92 +132,64 @@ public String getMethod() {
+                    ESFieldLabel.isLeaf.toString(), token.isLeaf(),
+                    ESFieldLabel.tenantId.toString(), token.getLocator().getTenantId()));
+        }
 
-        int statusCode = elasticsearchRestHelper.indexMetrics(metrics);
-        elasticsearchRestHelper.setBaseUrlForTestOnly("127.0.0.1:9200");
+        return sb.toString();
+    }
 
-        Assert.assertEquals(404, statusCode);
+    public void insertTokenDiscovery(List<Token> tokens, String indexName,
+                                     ElasticsearchRestHelper elasticsearchRestHelper) throws IOException {
+        if (tokens.size() == 0) return;
+        this.indexTokens(tokens, indexName, elasticsearchRestHelper);
     }
 
-    @Test
-    public void testCreateSingleRequest_WithNotNullMetricName() throws Exception {
-        String tenantId = "test1";
-        String metricName = "one.two.three.four.five";
-        Locator locator = Locator.createLocatorFromPathComponents(tenantId, metricName);
-        locatorMap.put(tenantId, new ArrayList<>());
-        locatorMap.get(tenantId).add(locator);
-
-        Metric metric =
-                new Metric(locator, 123456789L, 0, new TimeValue(1, TimeUnit.DAYS), UNIT);
-        elasticIO.insertDiscovery(metric);
-
-        int statusCode = elasticIO.elasticsearchRestHelper.refreshIndex(ELASTICSEARCH_INDEX_NAME_READ);
-
-        if(statusCode != 200) {
-            System.out.println(String.format("Refresh for %s failed with status code: %d",
-                    ELASTICSEARCH_INDEX_NAME_READ, statusCode));
-        }
+    @Override
+    protected void insertDiscovery(List<IMetric> metrics) throws IOException {
+        elasticIO.insertDiscovery(metrics);
 
-        List<SearchResult> results;
-        results = elasticIO.search(tenantId, metricName);
-        List<Locator> locators = locatorMap.get(tenantId);
-        assertEquals(locators.size(), results.size());
-        for (Locator l : locators) {
-            SearchResult entry =  new SearchResult(tenantId, l.getMetricName(), "");
-
-            boolean isFound = false;
-            for(SearchResult item : results){
-                if((StringUtils.isNotEmpty(item.getMetricName()) && item.getMetricName().equalsIgnoreCase(entry.getMetricName())) &&
-                        (StringUtils.isNotEmpty(item.getTenantId()) && item.getTenantId().equalsIgnoreCase(entry.getTenantId()))){
-                    isFound = true;
-                    break;
-                }
-            }
-            Assert.assertTrue(isFound);
-        }
+        Stream<Locator> locators = metrics.stream().map(IMetric::getLocator);
+        elasticTokensIO.insertDiscovery(Token.getUniqueTokens(locators)
+                .collect(toList()));
     }
 
     @Test
     public void testNoCrossTenantResults() throws Exception {
         List<SearchResult> results = elasticIO.search(TENANT_A, "*");
-        assertEquals(NUM_DOCS + 2, results.size());
+        assertEquals(NUM_DOCS, results.size());
```

Comment:
Changing the test expectations worries me.

Reply:
Just to make sure you have the context.. :-) This file is a complete redo. When you are comparing this file, please compare with the "master" branch file. There is zero change in the test cases except one and even that change is done to replace native client to the REST API call.

Reply:
Also, regarding change on expectations in the test case is tied to the test data got entered in the setup. But, at this point of time it's not relevant anymore.